thirdparty/pcap++/Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -26,11 +26,6 @@
 #endif
 #ifdef MAC_OS_X
 #include <net/if_dl.h>
-#endif
-
-#if defined(__linux__) && !defined(__GLIBC__)
-#include <linux/sysctl.h>
-#elif !defined(WIN32) && !defined(WINx64)
 #include <sys/sysctl.h>
 #endif

```

Comment:
It looks like we add an include `<sys/sysctl.h>` on mac, and remove all includes on linux, because this removal makes the whole block part of `#ifdef MAC_OS_X`. Was this your intention? Am I misunderstanding something?

Reply:
Yes, that was my intention. If you check the file `sysctl` call is only used on line [682](https://github.com/apache/nifi-minifi-cpp/blob/563848f760dfc679ad86fb476acb36e74142cdac/thirdparty/pcap%2B%2B/Pcap%2B%2B/src/PcapLiveDevice.cpp#L682) and [689](https://github.com/apache/nifi-minifi-cpp/blob/563848f760dfc679ad86fb476acb36e74142cdac/thirdparty/pcap%2B%2B/Pcap%2B%2B/src/PcapLiveDevice.cpp#L689) and those are only part of the MACOS build. The Jira ticket also mentions that this was the same fix in the newer version of the official Pcap++ repository in [this commit](https://github.com/seladb/PcapPlusPlus/commit/f474316b2fcdcd3b2d6ab6143ef4c3b2b5ec9ea8).